dotCMS/src/integration-test/java/com/dotcms/vanityurl/business/VanityUrlAPITest.java

```diff
@@ -60,271 +61,197 @@ public static void prepare() throws Exception {
-                            vanityURLContentlet.getLanguageId(),
-                            vanityURLContentlet.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                    ));
-            Assert.assertNotNull(vanityURLCached);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL, vanityURLCached.getVanityUrlId());
-            Assert.assertEquals("/testing"+currentTime, vanityURLCached.getUrl());
-
-            currentTime = System.currentTimeMillis();
-            Contentlet vanityURLContentletUpdated = contentletAPI.checkout(vanityURLContentlet.getInode(), user, false);
-            vanityURLContentletUpdated.setStringProperty("uri", "/testing"+currentTime);
-            vanityURLContentletUpdated = contentletAPI.checkin(vanityURLContentletUpdated, user, false);
-            filtersUtil.publishVanityUrl(vanityURLContentletUpdated);
-
-            vanityUrlAPI.getLiveCachedVanityUrl("/testing"+currentTime, defaultHost, defaultLanguageId, user);
-
-            vanityURLCached = vanityUrlCache.get(
-                    new CacheVanityKey(
-                            vanityURLContentlet.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                            vanityURLContentlet.getLanguageId(),
-                            vanityURLContentlet.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                    ));
-            Assert.assertNull(vanityURLCached);
-
-            vanityURLCached = vanityUrlCache.get(
-                    new CacheVanityKey(
-                            vanityURLContentletUpdated.getStringProperty(VanityUrlContentType.SITE_FIELD_VAR),
-                            vanityURLContentletUpdated.getLanguageId(),
-                            vanityURLContentletUpdated.getStringProperty(VanityUrlContentType.URI_FIELD_VAR)
-                    ));
-            Assert.assertNotNull(vanityURLCached);
-            Assert.assertNotEquals(VanityUrlAPI.CACHE_404_VANITY_URL, vanityURLCached.getVanityUrlId());
-            Assert.assertEquals("/testing"+currentTime, vanityURLCached.getUrl());
-
-        } finally {
-            contentletAPI.destroy(vanityURLContentlet, user, false );
-        }
+        long i = System.currentTimeMillis();
+        String title = "VanityURL" + i;
+        String site = defaultHost.getIdentifier();
+        String uri = "/test2_1_" + i;
+        String forwardTo = "/products/";
+        int action = 200;
+        int order = 1;
+
+        vanityURLContentlet = filtersUtil.createVanityUrl(title, site, uri,
+                forwardTo, action, order, defaultLanguage.getId());
+            
+        filtersUtil.publishVanityUrl(vanityURLContentlet);
+
+        Optional<CachedVanityUrl> vanityResolved = vanityUrlAPI.resolveVanityUrl(uri, defaultHost, defaultLanguage);
+
+        // we have vanity
+        assert(vanityResolved.isPresent());
+
+
+        // update the vanity
+        String newUri = "/testing"+System.currentTimeMillis();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "/testing" appears 15 times in this file; the first occurrence is on line 237](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5434315)